### PR TITLE
Introduce ScriptType enum

### PR DESCRIPTION
**Pull request by zeptin**
This is a potentially useful improvement from the upstream NBitcoin repository.

### CODE REVIEW

**Comment by dangershony** [comment]
Excellent PR.

**Comment by codingupastorm** [comment]
I like this, feel like it would be nice to use it somewhere before putting it in though.

**Comment by zeptin** [comment]
https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/4073

File: src/NBitcoin/Script.cs [hunk 1]
```diff
@@ -871,8 +888,10 @@ public BitcoinScriptAddress GetScriptAddress(Network network)
             return (BitcoinScriptAddress) this.Hash.GetAddress(network);
         }
 
+        [Obsolete("Use IsScriptType instead")]
```
Comment (fassadlr) [inline src/NBitcoin/Script.cs]: @zeptin if these are now obsolete, can we perhaps use the new ScriptType enum?

**Comment by codingupastorm** [comment]
If we just push the code in someone will have to update it eventually anyway 😉

**Pull request merged**